SDK gen breaks for GCP provider name

### What happened?

Building the SDK on GCP after #3188 fails because the schema name is different from what the bridge expects.

The bridge incorrectly uses `g.Info.Name` as the information for the Pulumi GCP provider name; that field is the _Terraform name_ which does not always agree with the Pulumi name and the build fails because the schema file cannot be opened.

### Example

```
PULUMI_HOME=/Users/guin/go/src/github.com/pulumi/pulumi-gcp/.pulumi PULUMI_CONVERT_EXAMPLES_CACHE_DIR=/Users/guin/go/src/github.com/pulumi/pulumi-gcp/.pulumi/examples-cache PULUMI_CONVERT=1 PULUMI_DISABLE_AUTOMATIC_PLUGIN_ACQUISITION=1 /Users/guin/go/src/github.com/pulumi/pulumi-gcp/bin/pulumi-tfgen-gcp nodejs --out sdk/nodejs/
warning: resource:google_apigee_keystores_aliases_key_cert_file.Docs received a non-zero custom value &{apigee_keysotres_aliases_key_cert_file.html.markdown []     false false} that is being ignored. Plugin Framework based providers do not yet support this feature.

Additional example conversion stats are available by setting COVERAGE_OUTPUT_DIR.

Provider:     google-beta
Success rate: NaN% (0/0)


General metrics:
	0 total resources containing 0 total inputs.
	0 total functions.

Argument metrics:
	0 argument descriptions were parsed from the upstream docs
	0 top-level input property descriptions came from an upstream attribute (as opposed to an argument). Nested arguments are not included in this count.
	0 of 0 resource inputs (NaN%) are missing descriptions in the schema

error: open provider/cmd/pulumi-resource-google-beta/schema.json: no such file or directory
Error: open provider/cmd/pulumi-resource-google-beta/schema.json: no such file or directory
Usage:
  /Users/guin/go/src/github.com/pulumi/pulumi-gcp/bin/pulumi-tfgen-gcp <LANGUAGE> [flags]

Flags:
  -d, --debug                 Enable debug logging
      --heap-profile string   Write a heap profile to this file
  -h, --help                  help for /Users/guin/go/src/github.com/pulumi/pulumi-gcp/bin/pulumi-tfgen-gcp
      --logtostderr           Log to stderr instead of to files
  -o, --out string            Emit the generated SDK to this directory
      --profile string        Write a CPU profile to this file
  -q, --quiet                 Suppress non-error output progress messages
      --skip-docs             Do not convert docs from TF Markdown
      --skip-examples         Do not convert examples from HCL
      --trace string          Write a Go runtime trace to this file
  -v, --verbose int           Enable verbose logging (e.g., v=3); anything >3 is very verbose

An error occurred: open provider/cmd/pulumi-resource-google-beta/schema.json: no such file or directory
make: *** [.make/generate_nodejs] Error 255
```

### Output of `pulumi about`

n/a

### Additional context

Introduced in https://github.com/pulumi/pulumi-terraform-bridge/pull/3188; unreleased as of today.

### Contributing

Vote on this issue by adding a 👍 reaction. 
To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already).

--- a/pkg/tfgen/generate.go
+++ b/pkg/tfgen/generate.go
@@ -1112,7 +1112,7 @@ func (g *Generator) Generate() (*GenerateSchemaResult, error) {
 		return g.UnstableGenerateFromSchema(genSchemaResult)
 	}
 	// Read the provider schema from file
-	schemaBytes, err := os.ReadFile(fmt.Sprintf("provider/cmd/pulumi-resource-%s/schema.json", g.info.Name))
+	schemaBytes, err := os.ReadFile(fmt.Sprintf("provider/cmd/pulumi-resource-%s/schema.json", g.pkg.Name().String()))
 	if err != nil {
 		return nil, err
 	}

